homeassistant/components/poolsense/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""The PoolSense integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+import poolsense
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client, update_coordinator
+
+from .const import DOMAIN
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the PoolSense component."""
+    # Make sure coordinator is initialized.
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up PoolSense from a config entry."""
+
+    await get_coordinator(hass, entry)
+
+    if not entry.unique_id:
+        hass.config_entries.async_update_entry(entry, unique_id=entry.data["email"])
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+
+    return unload_ok
+
+
+async def get_coordinator(hass, entry):
+    """Get the data update coordinator."""
+    if DOMAIN in hass.data:
+        return hass.data[DOMAIN]
+
+    async def async_get_cases():
```

Comment:
"cases" sounds like coronavirus reference

Reply:
Yes.Thanks. I have now corrected this.

Reply:
It is. I have renamed appropriately.